### DISCUSSION TITLE
Matching keyboard language to Settings language automatically

### DISCUSSION THREAD

**Discussion by distilledliquor** [valinet/ExplorerPatcher#597] [Ideas]
Users need to change the keyboard langauge at launching every applications if someone is using ExplorerPatcher as not English.
This is a traditional bug of Windows and some of programs can have a same issue but it appears on EP too unfortunately.
Is this could be fixed on later releases?

**Comment by Anixx** [comment]
Why would you need a keyboard layout at all when using Explorer Patcher? It has only one dialog without any text fields.

**Reply by valinet** [reply]
The OP is probably not talking about using EP's settings dialog, but rather when using Windows in general.

**Comment by valinet** [comment]
What's this "bug"? To be able to solve something, one needs to first understand it. I honestly do not understand what the bug is.

I know that the behavior in versions prior to Windows 8 was to have a default keyboard layout, and one could change it per application. Then, in Windows 8, they made it so that switching the language in one application switched it in all the other applications automatically, and this behavior has since been kept the default. As far as I know, one can restore the old behavior in Settings:

![image](https://user-images.githubusercontent.com/6503598/147386956-59c39cc6-b5d7-4e69-b8ea-4968e0449359.png)

Check to make sure you do not have that enabled by mistake. Other than that, if that's not the case, a good first step would be to properly explain this "traditional bug" and offer a way to clearly reproduce it.

**Reply by distilledliquor** [reply]
I made some of images to make you understand.
The problem is, that option doesn't make sense totally.

![3415](https://user-images.githubusercontent.com/54929046/147391327-6d856117-a83c-4896-a409-7a12979d2fb3.png)
![23421](https://user-images.githubusercontent.com/54929046/147391332-e1c55b6c-88a2-4e38-8e32-71007ba8ade5.png)
![2342333](https://user-images.githubusercontent.com/54929046/147391333-f8ef9e74-3d38-4ef0-9079-ecc6f365dc62.png)
![56342](https://user-images.githubusercontent.com/54929046/147391338-7952be92-0f97-4024-852f-3fbc1cc56ffb.png)
![3564352](https://user-images.githubusercontent.com/54929046/147391335-cbfabe4f-e5aa-4024-9a77-2d3ed5f9277b.png)

**Reply by valinet** [reply]
Yeah, idk, seems like a Windows bug to me. It seems IMEs and the Windows 10 taskbar in Windows 11 has some problems. I don’t really know, it has to be investigated. Maybe look here as well: https://github.com/valinet/ExplorerPatcher/issues/598

**Reply by valinet** [reply]
Yeah, seems to be a problem with the new Microsoft IME that's not really supported with the Windows 10 taskbar, or at least it needs some fixing up, idk. Try enabling the old IME as described [here](https://github.com/valinet/ExplorerPatcher/discussions/588#discussioncomment-1870928) and see if it still behaves weirdly. And please report back your findings. Hopefully it works with that and it is a workaround until someone figures out the newer one as well.
